### PR TITLE
Added global institutional and pipeline configs for sbc_sharc

### PR DESCRIPTION
**Pull request by lquayle88**
---
name: sbc_sharc
about: New institutional configuration profile for the ShARC cluster at the University of Sheffield
---

Please follow these steps before submitting your PR:

- [x] If your PR is a work in progress, include `[WIP]` in its title
- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `README.md` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

### CODE REVIEW

File: conf/sbc_sharc.config [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+
+// container engine
+
+singularity {
+
+  enabled = true
+  autoMounts = true
+  // cacheDir = '/<path>/<to>/<shared>/<image_cache>'
```
Comment (jfy133) [inline conf/sbc_sharc.config:37]: I would remove if not used, and you provide 'per user' instructions

File: conf/sbc_sharc.config [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  executor = 'sge'
+  penv = 'smp'
+  queue = { task.time <= 6.h ? 'shortint.q' : 'all.q' }
+  clusterOptions = { "-l rmem=${task.memory.toGiga()}G" }
+
+  // error and retry handling
+
+  errorStrategy = { task.exitStatus in [143,137,104,134,139,140] ? 'retry' : 'finish' }
+  maxRetries = 2
```
Comment (jfy133) [inline conf/sbc_sharc.config:56]: I would normally leave this on a per-pipeline basis, as exit codes can somtimes mean different things to different tools

File: docs/sbc_sharc.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+- [nf-co.re/sarek](https://nf-co.re/sarek)
+
+When using [`sbc_sharc.config`](../conf/sbc_sharc.config) with the pipelines listed above, the appropriate configuration file from the list below will be loaded automatically:
+
+- [atacseq sbc_sharc.config](../conf/pipeline/atacseq/sbc_sharc.config)
+- [chipseq sbc_sharc.config](../conf/pipeline/chipseq/sbc_sharc.config)
+- [rnaseq sbc_sharc.config](../conf/pipeline/rnaseq/sbc_sharc.config)
+- [sarek sbc_sharc.config](../conf/pipeline/sarek/sbc_sharc.config)
+
+The [`sbc_sharc.config`](../conf/sbc_sharc.config) configuration file might work with other nf-core pipelines as it stands but we cannot guarantee they will run without issue. We will be continuing to create, test and optimise configurations for new pipelines in the future.
```
Comment (jfy133) [inline docs/sbc_sharc.md:17]: I find this a bit confusing the start with the pipelines it's configured for, the base image should work with most and I owuld normally start with that and with a (generic) command  right at the beginning showing how to load the base config (i.e. the next section should go first)

**Review by jfy133 — CHANGES_REQUESTED** [review]
'Config' wise looks good! Just missing adding the pipeline-specific configs to the list near the bottom of the main README.

My main (opinionated) comment is about the documentation. It is really well written and I can see this would be very useful! However, I personally feel it's a bit too out of scope for the documentation about the configs themselves.

Namely sections 1-3 which is how to set up nextflow run rather than information about the configs. Really you would only be looking at the configs once you have nextflow set up and installed, so I think most of the sections would be better documented within your institutes documentation, as it's not nf-core nor config specific. 

More specifically, section 1-3 is not necessary here, section 4 is fine for this config document, but then step 5-6 is arguably an opinionated way of setting up a pipeline run (e.g. you're also assuming a pipeline has a samplesheet, but not all do), so I think that would also better be put on institutional documentation.

_that said_ if there is no suitable place at your institution for hosting such documentation, I guess it would be fine to keep here for now...

**Comment by lquayle88** [comment]
Thanks for reviewing my PR James and for the useful feedback. I have made (I think) all the changes you requested. I have removed most of the info from the documentation and will move this to a page on our own site for other users to refer to - I have left the reordered first section and the information relating to Singularity.

**Review by jfy133 — APPROVED** [review]
LGTM!

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Comment by jfy133** [comment]
@lquayle88 just need to run prettier on `docs/sbc_sharc.md` and then you can merge

**Comment by lquayle88** [comment]
Should be done now :)

**Pull request merged**